### DISCUSSION TITLE
Questions on rotation part of joint jacobian

### DISCUSSION THREAD

**Discussion by Cfather** [stack-of-tasks/pinocchio#2006] [Q&A]
Hi,

I have checked several previous issues on joint jacobian but I guess I'm still having trouble understanding it correctly. What does the rotation part (the last 3 rows) of the joint jacobian correspond to, specifically?

I'm working on friction on the end effector of a manipulator. So I would like to compute the jacobian of the roll, pitch, and yaw angle of the end effector frame (or yaw, pitch, roll, or whatever order). I'm not sure if the joint jacobian computed in Pinocchio can provide what I need here.

I would appreciate it very much if someone could help me with this conceptual question!

**Comment by jcarpent** [comment]
Pinocchio does not use roll, pitch, yaw convention to avoid singularity issues related to this parametrization of rotation.
The angular part associated with the 6D frame Jacobian corresponds to the variation of the orientation, in LOCAL or LOCAL_WORLD_ALIGNED frame convention, with respect to the variation of the joint configuration vector.

Certainly, you should provide a short code example (in Python if possible), in such a way we can help you further ;)
